### PR TITLE
obs-studio: 25.0.3 -> 25.0.8, use addOpenGLRunpath

### PR DESCRIPTION
**Pull request by elohmeier**
###### Motivation for this change
OBS was outdated and didn't make use of hardware-accelerated encoders like nvenc (Nvidia).

###### Things done
Updated to the latest version, added support for nvenc by configuring the addOpenGLRunpath fixup.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by elohmeier** [comment]
Restored original formatting to ease reviewing.

**Comment by elohmeier** [comment]
@flokli could you take a look at this PR? It's similar to the ffmpeg PR you've reviewed.

File: pkgs/applications/video/obs-studio/default.nix [hunk 1]
```diff
@@ -80,6 +81,10 @@ in mkDerivation rec {
         --prefix "LD_LIBRARY_PATH" : "${xorg.libX11.out}/lib:${vlc}/lib"
   '';
 
+  postFixup = stdenv.lib.optionalString stdenv.isLinux ''
+      addOpenGLRunpath $out/lib/lib*.so
```
Comment (flokli) [inline pkgs/applications/video/obs-studio/default.nix:85]: This probably needs a bit more investigation:
```
❯ ag -a dlopen result/lib
Binary file libobsglad.so matches.
Binary file libobsglad.so.0 matches.
Binary file libobs.so matches.
Binary file libobs.so.0 matches.
Binary file obs-plugins/linux-decklink.so matches.
Binary file obs-plugins/obs-ffmpeg.so matches.
Binary file obs-plugins/vlc-video.so matches.
```
Did you check `obs-ffmpeg.so` only `dlopen`s `libavcodec.so`, which should be fixed by https://github.com/NixOS/nixpkgs/pull/87588?

Comment (elohmeier) [reply pkgs/applications/video/obs-studio/default.nix:85]: I've just tested against the fixed ffmpeg from #87968 without the addOpenGLRunpath change in obs.
The ffmpeg change is not fixing it for obs, I get the following log output when starting obs:
`error: os_dlopen(libnvidia-encode.so.1->libnvidia-encode.so.1): libnvidia-encode.so.1: cannot open shared object file: No such file or directory`.

Comment (flokli) [reply pkgs/applications/video/obs-studio/default.nix:85]: So something inside `obs` tries to `dlopen` `libnvidia-encode.so.1` directly, okay.

I did a `ag -a libnv result/`:
```
Binary file lib/obs-plugins/obs-ffmpeg.so matches.
```

it looks like `lib/obs-plugins/obs-ffmpeg.so` is what's trying to open this - so we should need to patch this, which isn't covered by your glob `$out/lib/lib*.so`? :confused:

Comment (elohmeier) [reply pkgs/applications/video/obs-studio/default.nix:85]: I've added the call for the obs-plugins folder.

Comment (flokli) [reply pkgs/applications/video/obs-studio/default.nix:85]: I'm not sure if the other line is still necessary as well, but it's probably fine.

**Pull request merged**